Transcode

romashorodok opened the pull request:
Make audio chunks based on provided audio file

romashorodok committed:
feat: Added seeding for models.

romashorodok committed:
feat: Add assets for seeding.

romashorodok committed:
feat: Produce topic for transcode service.

romashorodok committed:
feat: Added transcode service.

romashorodok committed:
fix: Format and don't transcode on empty file path

romashorodok committed:
fix: Add timeout while reading a message.

romashorodok committed:
feat: Poll messages instead of reading.

romashorodok committed:
feat: Update segment bucket on success audio processing

romashorodok committed:
feat: Check if manifest exists.

romashorodok committed:
feat: Added audio file seeding

romashorodok committed:
ref: rename typo

romashorodok committed:
feat: Added blacklist for seeding files and change kafka.

romashorodok committed:
ref: delete assets.

ionhrytsku commented on app/Observers/AudioObserver.php:
```diff
@@ -0,0 +1,89 @@
+        /* @var Filesystem $disk */
+        $disk = $fs->disk('minio.segment');
+
+        $this->disk = $disk;
```
The $disk declaration can be moved to ServiceProvider. Thus, the class would depend only on `Filesystem` variable, not on FilesystemManager. And since Filesystem instance seems like reusable, it should not be reinitiated in code.

ionhrytsku commented on app/Observers/AudioObserver.php:
```diff
@@ -0,0 +1,89 @@
+        if ($manifest != null && $manifest != '') {
+            $exists = $this->disk->fileExists($bucket . "/" . $manifest);
+
+            if (!$exists) {
```
Both conditions can be moved in 1 condition block. No need to cascade them.

ionhrytsku commented on app/Observers/AudioObserver.php:
```diff
@@ -0,0 +1,89 @@
+        $manifest = $segmentBucket['manifest_file'];
+        $bucket = $segmentBucket['bucket'];
+
+        if ($manifest != null && $manifest != '') {
```
Can be replaced with `!empty($manifest)`

ionhrytsku commented on app/Observers/AudioObserver.php:
```diff
@@ -0,0 +1,89 @@
+        if ($file && $file != '') {
+            $sameSegments = $segmentBucket->segmentsOf($file);
+
+            if ($sameSegments) return;
```
Please follow the PSR ruels. Always write control structures with braces.
https://www.php-fig.org/psr/psr-12/#5-control-structures

ionhrytsku commented on app/Traits/TranscodeAudio.php:
```diff
@@ -0,0 +1,54 @@
+use League\Flysystem\Filesystem;
+use League\Flysystem\FilesystemException;
+
+trait TranscodeAudio
```
Traits is a powerful mechanic in PHP. But main purpose of trait - is to extend some basic functionality of existing class. Meaning, that new functionality MUST be logically related to the initial functionality of the class. E.g. Auth trait for Controllers.
In this case it looks more like this trait should be a service, which would be injected into Observer, not used as trait.

ionhrytsku reviewed (commented):
Sorry, haven't check the go files. Let's just hope they are fine ))

ionhrytsku reviewed (commented):
Sorry, haven't check the go files. Let's just hope they are fine ))

romashorodok committed:
Merge branch 'upload' into transcode

romashorodok committed:
ref: Remove hardcoded `$disk` variable. Inject directly instead

romashorodok committed:
feat: Move `TranscodeAudio` trait into own service

romashorodok committed:
ref: Simplify empty conditions.

romashorodok committed:
ref: Delete `TranscodeAudio` trait

romashorodok committed:
feat: Transcode if manifest or bucket empty

romashorodok committed:
fix: Fix ffmpeg process crash.

  When func completes, it's garbage collect process and it's break all
  process chain

romashorodok committed:
fix: Fix broken upload logic

romashorodok committed:
feat: Added error loging for ffmpeg
